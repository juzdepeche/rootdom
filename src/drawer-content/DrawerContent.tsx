import React, { useState } from 'react';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
import { TextCheckBox } from './TextCheckBox';
import { Faction } from '../../App';

type DrawerContentProps = {
	factions: Faction[];
	onFactionToggle: (factionName: string, isSelected: boolean) => void;
};

export const DrawerContent = ({
	factions,
	onFactionToggle
}: DrawerContentProps) => {
	return (
		<View style={styles.container}>
			<Text style={styles.header}>Factions</Text>
			<View style={styles.checkboxContainer}>
				{factions.map((faction) => (
					<TextCheckBox
						key={faction.name}
						text={faction.name}
						onFactionToggle={onFactionToggle}
					></TextCheckBox>
				))}
			</View>
		</View>
	);
};
const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#fff',
		alignItems: 'center',
		marginTop: 20
	},
	header: {
		fontSize: 25,
		fontWeight: 'bold',
		color: '#364f6b',
		marginBottom: 40,
		paddingTop: 40
	},
	checkboxContainer: {
		width: '80%',
		alignItems: 'center'
	}
});
