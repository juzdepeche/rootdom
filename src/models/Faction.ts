import images from '../../assets/images';

export class Faction {
	public name: string;
	public value: number;
	public color: string;
	public image;

	constructor(name: string, value: number, color: string, image?) {
		this.name = name;
		this.value = value;
		this.color = color;
		this.image = image;
	}
}

export const Factions = [
	new Faction('Marquise de Cat', 10, '#FD5200', images.cat),
	new Faction('Underground Duchy', 8, '#D7A872', images.taupe),
	new Faction('Eyrie Dynasties', 7, '#4774B7', images.bird),
	new Faction('Vagabond', 5, '#FCFEFD', images.vagabond),
	new Faction('Riverfolk Company', 5, '#61DDFB', images.otter),
	new Faction('Woodland Alliance', 3, '#59BA53', images.mouse),
	new Faction('Corvid Consipiracy', 3, '#290D7E', images.raven),
	new Faction('Lizard Cult', 2, '#f1f435', images.lizard)
];

export const GetViableGameSum = (playerCount:number) => {
	switch (playerCount) {
		case 2:
			return 17;
		case 3:
			return 18;
		case 4:
			return 21;
		case 5:
			return 25;
		case 6:
			return 28;
		default:
			return 0;
	}
}

export const GetBigFactionsSum = (factions:Faction[], playerCount:number) => {
	let sum = 0;
	factions.sort((a, b) => a.value > b.value ? -1 : 1)
	for (let i = 0; i < playerCount; i++) {
		sum += factions[i].value;
	}
	return sum;
}