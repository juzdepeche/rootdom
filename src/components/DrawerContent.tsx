import React from 'react';
import { ScrollView, StyleSheet, Text, View } from 'react-native';
import { Faction } from '../models/Faction';
import { TextCheckBox } from './TextCheckBox';

type DrawerContentProps = {
	factions: Faction[];
	onFactionToggle: (factionName: string, isSelected: boolean) => void;
};

export const DrawerContent = ({
	factions,
	onFactionToggle
}: DrawerContentProps) => {
	return (
		<View style={styles.container}>
			<Text style={styles.header}>Factions</Text>
			<ScrollView style={styles.scrollViewContainer}>
				<View style={styles.checkboxContainer}>
					<View style={styles.centeredCheckboxContainer}>
						{factions.map((faction) => (
							<TextCheckBox
								key={faction.name}
								text={faction.name}
								selected={faction.available}
								onFactionToggle={onFactionToggle}
							></TextCheckBox>
						))}
					</View>
				</View>
			</ScrollView>
		</View>
	);
};
const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#fff',
		alignItems: 'center',
		marginTop: 20
	},
	header: {
		fontSize: 25,
		fontWeight: 'bold',
		color: '#364f6b'
	},
	scrollViewContainer: {
		width: '100%',
		marginVertical: 40
	},
	checkboxContainer: {
		width: '100%',
		alignItems: 'center'
	},
	centeredCheckboxContainer: {
		width: '80%'
	}
});
