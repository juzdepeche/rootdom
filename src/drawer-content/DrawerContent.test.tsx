import React from 'react';
import renderer from 'react-test-renderer';
import { Factions } from '../models/Faction';
import { DrawerContent } from './DrawerContent';

describe('<DrawerContent />', () => {
	it('renders correctly', () => {
		const drawerContent = renderer
			.create(
				<DrawerContent
					factions={[...Factions]}
					onFactionToggle={() => null}
				></DrawerContent>
			)
			.toJSON();
		expect(drawerContent).toMatchSnapshot();
	});
});
