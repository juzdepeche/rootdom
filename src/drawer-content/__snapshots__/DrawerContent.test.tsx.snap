// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerContent /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "marginTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#364f6b",
        "fontSize": 25,
        "fontWeight": "bold",
        "marginBottom": 40,
        "paddingTop": 40,
      }
    }
  >
    Factions
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "width": "80%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#f6f6f6",
          "borderRadius": 20,
          "flexDirection": "row",
          "marginBottom": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <View
        pointerEvents="auto"
      >
        <RNCCheckbox
          forwardedRef={null}
          onChange={[Function]}
          onValueChange={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 32,
                "width": 32,
              },
              undefined,
            ]
          }
          value={true}
        />
      </View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#364f6b",
              "fontSize": 20,
              "fontWeight": "bold",
              "marginLeft": 20,
            },
          ]
        }
      >
        Marquise de Cat
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#f6f6f6",
          "borderRadius": 20,
          "flexDirection": "row",
          "marginBottom": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <View
        pointerEvents="auto"
      >
        <RNCCheckbox
          forwardedRef={null}
          onChange={[Function]}
          onValueChange={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 32,
                "width": 32,
              },
              undefined,
            ]
          }
          value={true}
        />
      </View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#364f6b",
              "fontSize": 20,
              "fontWeight": "bold",
              "marginLeft": 20,
            },
          ]
        }
      >
        Underground Duchy
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#f6f6f6",
          "borderRadius": 20,
          "flexDirection": "row",
          "marginBottom": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <View
        pointerEvents="auto"
      >
        <RNCCheckbox
          forwardedRef={null}
          onChange={[Function]}
          onValueChange={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 32,
                "width": 32,
              },
              undefined,
            ]
          }
          value={true}
        />
      </View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#364f6b",
              "fontSize": 20,
              "fontWeight": "bold",
              "marginLeft": 20,
            },
          ]
        }
      >
        Eyrie Dynasties
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#f6f6f6",
          "borderRadius": 20,
          "flexDirection": "row",
          "marginBottom": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <View
        pointerEvents="auto"
      >
        <RNCCheckbox
          forwardedRef={null}
          onChange={[Function]}
          onValueChange={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 32,
                "width": 32,
              },
              undefined,
            ]
          }
          value={true}
        />
      </View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#364f6b",
              "fontSize": 20,
              "fontWeight": "bold",
              "marginLeft": 20,
            },
          ]
        }
      >
        Vagabond
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#f6f6f6",
          "borderRadius": 20,
          "flexDirection": "row",
          "marginBottom": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <View
        pointerEvents="auto"
      >
        <RNCCheckbox
          forwardedRef={null}
          onChange={[Function]}
          onValueChange={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 32,
                "width": 32,
              },
              undefined,
            ]
          }
          value={true}
        />
      </View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#364f6b",
              "fontSize": 20,
              "fontWeight": "bold",
              "marginLeft": 20,
            },
          ]
        }
      >
        Riverfolk Company
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#f6f6f6",
          "borderRadius": 20,
          "flexDirection": "row",
          "marginBottom": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <View
        pointerEvents="auto"
      >
        <RNCCheckbox
          forwardedRef={null}
          onChange={[Function]}
          onValueChange={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 32,
                "width": 32,
              },
              undefined,
            ]
          }
          value={true}
        />
      </View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#364f6b",
              "fontSize": 20,
              "fontWeight": "bold",
              "marginLeft": 20,
            },
          ]
        }
      >
        Woodland Alliance
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#f6f6f6",
          "borderRadius": 20,
          "flexDirection": "row",
          "marginBottom": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <View
        pointerEvents="auto"
      >
        <RNCCheckbox
          forwardedRef={null}
          onChange={[Function]}
          onValueChange={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 32,
                "width": 32,
              },
              undefined,
            ]
          }
          value={true}
        />
      </View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#364f6b",
              "fontSize": 20,
              "fontWeight": "bold",
              "marginLeft": 20,
            },
          ]
        }
      >
        Corvid Consipiracy
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#f6f6f6",
          "borderRadius": 20,
          "flexDirection": "row",
          "marginBottom": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <View
        pointerEvents="auto"
      >
        <RNCCheckbox
          forwardedRef={null}
          onChange={[Function]}
          onValueChange={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "height": 32,
                "width": 32,
              },
              undefined,
            ]
          }
          value={true}
        />
      </View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#364f6b",
              "fontSize": 20,
              "fontWeight": "bold",
              "marginLeft": 20,
            },
          ]
        }
      >
        Lizard Cult
      </Text>
    </View>
  </View>
</View>
`;
